openssh: update to 10.0p2

artkiver opened the pull request:
* closes: https://trac.macports.org/ticket/72317

#### Description

<!-- Note: it is best to make pull requests from a branch rather than from master -->

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
<!-- Triple-click and copy the next line and paste it into your shell. It will copy your OS and Xcode version to the clipboard. Paste it here replacing this section.
sh -c 'echo "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion) $(uname -m)"; xcode=$(xcodebuild -version 2>/dev/null); if [ $? == 0 ]; then echo "$(echo "$xcode" | awk '\''NR==1{x=$0}END{print x" "$NF}'\'')"; else echo "Command Line Tools $(pkgutil --pkg-info=com.apple.pkg.CLTools_Executables | awk '\''/version:/ {print $2}'\'')"; fi' | tee /dev/tty | pbcopy
-->
macOS 15.4.1 24E263 arm64
Command Line Tools 16.3.0.0.1.1742442376

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? <!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

barracuda156 commented:
Builds fine for me on powerpc.

artkiver commented:
> Builds fine for me on powerpc.

Good to know! I'm not sure what's up with the CI checks failing. Something similar happened with my got-portable0.111 PR.

I have been planning to try (for at least the third time, hopefully that will be the charm?) to get mpbb running locally to see if I can recreate whatever GitHub CI may be tripping over? I've been pretty busy (my car was totaled recently, and since I am homeless/sleep in my car that means lots of hotel expenses and rental cars while I test drive potential replacements) so I am not sure when I will get to that. tbh, I was kind of hoping someone else might rework the sshd-session.c and ssh-agent.c patches so MacPorts still leverages launchd and the sandboxing framework, but my first attempts mentioned in the Trac issue cause different failures so I removed them for the time being.

Chiming in now, because I noticed LibreSSL 4.1.0 just got released, so I am prepping a PR for that next.

neverpanic commented on net/openssh/Portfile:
```diff
@@ -5,8 +5,9 @@ PortSystem          1.0
-version             9.9p2
-revision            1
+version             10.0p2
+distname            openssh-10.0p1
```
CI fails because the `ssh-copy-id` subport re-sets `$distname` to `openssh-10.0p2` in line 264, and that file isn't on the CDN.

artkiver reviewed (commented):
Thank you for spotting this!

reneeotten commented:
> Thank you for spotting this!

can you please fix that issue then so we might merge the PR?

artkiver commented:
> > Thank you for spotting this!
> 
> can you please fix that issue then so we might merge the PR?

My apologies for the delay, without getting into the hellscape that has been my personal life for the past week or so.

I already went through the GitHub "review" process and I thought I approved @neverpanic's change for whatever it was worth? It seemed as if GitHub's process was still waiting on something to the effect of "waiting on a reviewer with write access" (paraphrasing as I don't remember the precise language used when I last looked at this) the last time I had checked.

I can modify the Portfile in my local branch with Clemens' changes and push that instead if that methodology is preferable?

neverpanic commented:
> I already went through the GitHub "review" process and I thought I approved (or maybe "resolved" in their terminology?) @neverpanic's change for whatever it was worth? It seemed as if GitHub's process was still waiting on something to the effect of "waiting on a reviewer with write access" (paraphrasing as I don't remember the precise language used when I last looked at this) the last time I had checked.

I haven't made a change, I merely added a comment.


> I can modify the Portfile in my local branch with Clemens' changes and push that instead if that methodology is preferable?

Yes, please.

artkiver commented:
> > I already went through the GitHub "review" process and I thought I approved (or maybe "resolved" in their terminology?) @neverpanic's change for whatever it was worth? It seemed as if GitHub's process was still waiting on something to the effect of "waiting on a reviewer with write access" (paraphrasing as I don't remember the precise language used when I last looked at this) the last time I had checked.
> 
> I haven't made a change, I merely added a comment.
> 
> > I can modify the Portfile in my local branch with Clemens' changes and push that instead if that methodology is preferable?
> 
> Yes, please.

My apologies for the confusion and delays!

Hopefully that last push does the trick? It looks as if it at least passed the GitHub Actions checks.

neverpanic reviewed (approved):
Looks good to me now. Should we add a note saying that DSA support was completely removed for those that still have DSA keys and will now see failures, or do we let them figure this out on their own?

artkiver commented:
> Looks good to me now. Should we add a note saying that DSA support was completely removed for those that still have DSA keys and will now see failures, or do we let them figure this out on their own?

I figure it doesn't hurt to let folks know?

I added some details to that effect with notes-append and cleaned up another DSA reference I had previously overlooked with the most recent push.

artkiver commented:
Thank you @neverpanic !
